lib/workers/repository/updates/generate.ts

```diff
@@ -304,6 +304,10 @@ export function generateBranchConfig(
   config.blockedByPin = config.upgrades.every(
     (upgrade) => upgrade.blockedByPin
   );
+  config.compatibility = Object.assign(
+    {},
+    ...config.upgrades.map((upgrade) => upgrade.compatibility)
```

Comment:
Can you expand the testing so that it includes:
- Duplicate compatibility (e.g. two upgrades each with same compatibility key)
- Multiple compatibility keys (npm, yarn, etc)
- Some upgrades with no compatibility (undefined)
?

Reply:
@rarkins Should we combine `Duplicate compatibility` with `||`? Since `range-set  ::= range ( logical-or range ) *`, it'd be a valid semver range.

Reply:
@rarkins How `Duplicate compatibility` should be handled?

Reply:
In case of duplicate, let's just take the last/latest one in the array.

Reply:
@rarkins Then this PR is ready for review.